src/libappimage/libappimage.c

```diff
@@ -397,48 +397,191 @@ gchar **squash_get_matching_files_install_icons_and_mime_data(sqfs* fs, char* pa
     return (gchar **) g_ptr_array_free(array, FALSE);
 }
 
-/* Loads a desktop file from squashfs into an empty GKeyFile structure.
- * FIXME: Use sqfs_lookup_path() instead of g_key_file_load_from_squash()
- * should help for performance. Please submit a pull request if you can
- * get it to work.
+
+
+/**
+ * Lookup a given <path> in <fs>. If the path points to a symlink it is followed until a regular file is found.
+ * This method is aware of symlink loops and will fail properly in such case.
+ * @param fs
+ * @param path
+ * @param inode [RETURN PARAMETER] will be filled with a regular file inode. It cannot be NULL
+ * @return succeed true if the file is found, otherwise false
  */
-gboolean g_key_file_load_from_squash(sqfs *fs, char *path, GKeyFile *key_file_structure, gboolean verbose) {
-    sqfs_traverse trv;
-    gboolean success = true;
-    sqfs_err err = sqfs_traverse_open(&trv, fs, sqfs_inode_root(fs));
+bool sqfs_lookup_path_resolving_symlinks(sqfs* fs, char* path, sqfs_inode* inode) {
+    g_assert(fs != NULL);
+    g_assert(inode != NULL);
+
+    bool found = false;
+    sqfs_inode root_inode;
+    sqfs_err err = sqfs_inode_get(fs, &root_inode, fs->sb.root_inode);
     if (err != SQFS_OK) {
-        fprintf(stderr, "sqfs_traverse_open error\n");
+        g_warning("sqfs_inode_get root inode error\n");
```

Comment:
Shouldn't this be guarded by an `#ifdef` like the other logging?

Reply:
glib verbosity can be configured by the environmental variable `G_MESSAGES_DEBUG`. So it can be changed at runtime.

If you don't want to have this message shown by default we could use `g_info` instead. But I guess that this is something that the user should see at first glance.

Reply:
I surrounded the g_warning calls with the `#ifdef #endif` so the old logging behavior will be preserved.